agent_job_history.go

```diff
@@ -64,3 +54,21 @@ func agentJobHistory(client gocd.Client, agents []*gocd.Agent, agentJobResultCou
 	}
 	return nil
 }
+
+func NewAgentJobHistoryCache() *AgentJobHistoryCache {
+	return &AgentJobHistoryCache{}
+}
+
+func (c *AgentJobHistoryCache) Get(agentHostname string) int {
+	if job, ok := c.AgentJob[agentHostname]; ok {
+		return job
+	}
+	return 0
+}
+
+func (c *AgentJobHistoryCache) Set(agentHostname string, job int) {
```

Comment:
since the set/get only does what a map does (and theres no interface involved?), perhaps type the map directly `type AgentJobHistoryCache map[string]int`

having Get/Set method names are considered [bad practice](https://golang.org/doc/effective_go.html#Getters) in Go